browser/ui/webui/brave_rewards_internals_ui.cc

```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/brave_rewards_internals_ui.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "brave/components/brave_rewards/browser/rewards_service_factory.h"
+#include "brave/components/brave_rewards/common/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/web_ui.h"
+#include "content/public/browser/web_ui_message_handler.h"
+
+#if !defined(OS_ANDROID)
+#include "brave/components/brave_rewards/resources/grit/brave_rewards_generated_map.h"
+#include "brave/components/brave_rewards/resources/grit/brave_rewards_resources.h"
+#else
+#include "components/brave_rewards/settings/resources/grit/brave_rewards_settings_generated_map.h"
+#include "components/grit/components_resources.h"
+#include "components/grit/components_scaled_resources.h"
+#endif
+
+namespace {
+
+class RewardsInternalsDOMHandler : public content::WebUIMessageHandler {
+ public:
+  RewardsInternalsDOMHandler();
+  ~RewardsInternalsDOMHandler() override;
+
+  void Init();
+
+  // WebUIMessageHandler implementation.
+  void RegisterMessages() override;
+
+ private:
+  bool IsRewardsEnabled() const;
+  void HandleGetRewardsEnabled(const base::ListValue* args);
+  void HandleGetRewardsInternalsInfo(const base::ListValue* args);
+  void OnGetRewardsInternalsInfo(
+      std::unique_ptr<brave_rewards::RewardsInternalsInfo> info);
+  void OnPreferenceChanged();
+
+  brave_rewards::RewardsService* rewards_service_;  // NOT OWNED
+  Profile* profile_;
+  std::unique_ptr<PrefChangeRegistrar> pref_change_registrar_;
+  base::WeakPtrFactory<RewardsInternalsDOMHandler> weak_ptr_factory_;
+
+  DISALLOW_COPY_AND_ASSIGN(RewardsInternalsDOMHandler);
+};
+
+RewardsInternalsDOMHandler::RewardsInternalsDOMHandler()
+    : rewards_service_(nullptr), profile_(nullptr), weak_ptr_factory_(this) {}
+
+RewardsInternalsDOMHandler::~RewardsInternalsDOMHandler() {}
+
+void RewardsInternalsDOMHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      "brave_rewards_internals.getRewardsEnabled",
+      base::Bind(&RewardsInternalsDOMHandler::HandleGetRewardsEnabled,
```

Comment:
sorry, this should also be Repeating. `base::Bind` is being deprecated so we should use Once/Repeating